cargo-dist/src/backend/ci/github.rs

```diff
@@ -164,13 +178,50 @@ impl GithubCiInfo {
         let pr_run_mode = dist.pr_run_mode;
 
         let tap = dist.tap.clone();
-        let plan_jobs = dist.plan_jobs.clone();
-        let local_artifacts_jobs = dist.local_artifacts_jobs.clone();
-        let global_artifacts_jobs = dist.global_artifacts_jobs.clone();
-        let host_jobs = dist.host_jobs.clone();
+
+        let mut job_permissions = dist.github_custom_job_permissions.clone();
+        // use publish jobs default to elevated priviledges
+        for name in &dist.user_publish_jobs {
+            job_permissions.entry(name.clone()).or_insert_with(|| {
+                [
+                    ("id-token".to_owned(), "write".to_owned()),
+                    ("packages".to_owned(), "write".to_owned()),
```

Comment:
This drops attestations from the root permisisons, breaking any build steps that need to create attestations. eg, with default settings and attestations on:

```diff
 permissions:
-  contents: write
-  id-token: write
-  attestations: write
+  "contents": "write"
+  "id-token": "write"
+  "packages": "write"
```

Reply:
yikes, good catch!